lib/fluent/plugin_helper/timer.rb

```diff
@@ -76,8 +76,12 @@ def on_timer
         rescue => e
           @log.error "Unexpected error raised. Stopping the timer.", title: @title, error: e
           @log.error_backtrace
-          self.detach
+          detach
           @log.error "Timer detached.", title: @title
+        ensure
+          if attached?
+            detach unless @repeating
```

Comment:
Is `@repeating` an instance variable of `TimerWatcher`?
I think it's too internal to use it, and it looks too fragile.
We should store `repeat` into our own instance variable in `#initialize` and use it.